crates/wasmtime/src/func.rs

```diff
@@ -1063,31 +1084,70 @@ impl Func {
     }
 
     #[inline]
-    pub(crate) fn caller_checked_anyfunc(
+    pub(crate) fn caller_checked_func_ref(
         &self,
-        store: &StoreOpaque,
+        store: &mut StoreOpaque,
     ) -> NonNull<VMCallerCheckedFuncRef> {
-        store.store_data()[self.0].export().anyfunc
+        let func_data = &mut store.store_data_mut()[self.0];
+        if let Some(func_ref) = func_data.in_store_func_ref {
+            func_ref
+        } else {
+            let func_ref = func_data.export().anyfunc;
+            unsafe {
+                if func_ref.as_ref().wasm_call.is_none() {
+                    let func_ref = store.func_refs().push(func_ref.as_ref().clone());
+                    store.store_data_mut()[self.0].in_store_func_ref = Some(func_ref);
+                    store.fill_func_refs();
```

Comment:
I know we split this into a two-phase operation, but given that the push/fill is always back-to-back it may be better to just remove this `fill_func_refs` perhaps? (folding it into the push operation that is)

Reply:
Ah unfortunately the borrows don't really work out becuase `store` is already borrowed immutably for the `func_refs()` method and so we can't also get and pass in the store's module registry.